tests/e2e/fixtures/test_data.py

```
@@ -0,0 +1,54 @@
+    
+    @staticmethod
+    def create_sample_json():
+        """Create a sample JSON file for testing uploads."""
+        json_data = {
+            "users": [
+                {"id": 1, "name": "John Doe", "email": "john@example.com"},
+                {"id": 2, "name": "Jane Smith", "email": "jane@example.com"}
+            ]
+        }
+        
+        temp_file = tempfile.NamedTemporaryFile(mode='w', suffix='.json', delete=False)
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)